meta/src/html.rs

```diff
@@ -8,18 +8,29 @@ use std::{cell::RefCell, rc::Rc};
 pub struct HtmlContext {
     lang: Rc<RefCell<Option<TextProp>>>,
     dir: Rc<RefCell<Option<TextProp>>>,
+    class: Rc<RefCell<Option<TextProp>>>,
 }
 
 impl HtmlContext {
     /// Converts the `<html>` metadata into an HTML string.
     pub fn as_string(&self) -> Option<String> {
-        match (self.lang.borrow().as_ref(), self.dir.borrow().as_ref()) {
-            (None, None) => None,
-            (Some(lang), None) => Some(format!(" lang=\"{}\"", lang.get())),
-            (None, Some(dir)) => Some(format!(" dir=\"{}\"", dir.get())),
-            (Some(lang), Some(dir)) => {
+        match (self.lang.borrow().as_ref(), self.dir.borrow().as_ref(), self.class.borrow().as_ref()) {
```

Comment:
This is going to grow exponentially if we add any more attributes... With 4 conditions it was fine but this is a lot of duplication. Can you try a refactor?